src/amo/sagas/collections.js

```diff
@@ -213,25 +229,77 @@ export function* updateCollection({
       `/${lang}/${clientApp}/collections/${user}/${effectiveSlug}/`
     ));
 
+    const slugWasEdited = slug && slug !== collectionSlug;
     if (!slugWasEdited) {
       // Invalidate the stored collection object. This will force each
       // component to re-fetch the collection. This is only necessary
       // when the slug hasn't changed.
       yield put(deleteCollectionBySlug(effectiveSlug));
     }
   } catch (error) {
-    log.warn(`Failed to update collection: ${error}`);
+    log.warn(`Failed to ${action} collection: ${error}`);
     yield put(errorHandler.createErrorAction(error));
     yield put(finishFormOverlaySubmit(formOverlayId));
   }
 }
 
+export function* createCollection({
+  payload: {
+    errorHandlerId,
+    defaultLocale,
+    description,
+    formOverlayId,
+    name,
+    slug,
+    user,
+  },
+}: CreateCollectionAction,
+_modifyCollection: typeof modifyCollection = modifyCollection,
```

Comment:
If this gets too bonkers then you could just use one saga for both instead of defining separate create and update sagas. What I mean is instead of this:

```js
export default function* collectionsSaga() {
  yield takeLatest(CREATE_COLLECTION, createCollection);
  yield takeLatest(UPDATE_COLLECTION, updateCollection);
}
```

...just use the same handler for both actions:

```js
export default function* collectionsSaga() {
  yield takeLatest(
    [CREATE_COLLECTION, UPDATE_COLLECTION], 
    modifyCollection
  );
}
```
The `modifyCollection` handler could look at `action.type` to know if it got a create/update action.

I don't know if this would help with the testing problem or not, just something that came to mind.